exonum/src/blockchain/mod.rs

```diff
@@ -286,6 +286,19 @@ impl Blockchain {
         crypto::hash(&vec)
     }
 
+    #[doc(hidden)]
+    pub fn broadcast_raw_transaction(&self, tx: RawTransaction) -> Result<(), failure::Error> {
```

Comment:
Where do you plan to call this method from? Asking to better understand possible invariants we could enforce. E.g., the method is only called from service code, it could make sense to check that the service can only send transactions to itself (not saying that it will necessarily cover all use cases, but it sounds like an understandable restriction).

Reply:
This method is called by internal code of Java Bindings (see NodeContext::submit method [here](https://github.com/exonum/exonum-java-binding/blob/new_message_format/exonum-java-binding-core/rust/src/proxy/node.rs#L67). User will be able to submit some transaction from inside `Service#afterCommit` method, for example.

I don't think we need any restrictions in 0.10, as they probably make some of use cases impossible to implement.

Reply:
@vitvakatu does it make sense to add the check like this ?
```rust
if self.service_map.contains_key(&service_id) {
```

Reply:
@aleksuss i.e. with positive-if instead of negative-if? I've added this check in the latest commit

Reply:
@vitvakatu yeah i've seen it. It's ok.